File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/webhook/WebHookUpdateProcessor.java
package com.github.bumblebee.webhook;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import telegram.domain.Update;
import telegram.polling.HandlerRegistry;
import telegram.polling.TelegramUpdateConsumer;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

@Component
public class WebHookUpdateProcessor {

    private final TelegramUpdateConsumer updateConsumer;
    private final ExecutorService executor;

    @Autowired
    public WebHookUpdateProcessor(HandlerRegistry handlerRegistry) {
        this.updateConsumer = new TelegramUpdateConsumer(handlerRegistry);
        executor = Executors.newSingleThreadExecutor();
    }

    public void process(Update update) {
        executor.submit(() -> updateConsumer.accept(update));
    }
}

File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/webhook/WebHookUpdateProcessor.kt
package com.github.bumblebee.webhook

import com.github.bumblebee.bot.consumer.TelegramUpdateRouter
import com.github.telegram.domain.Update
import org.springframework.stereotype.Component
import java.util.concurrent.Executors

@Component
class WebHookUpdateProcessor(private val updateConsumer: TelegramUpdateRouter) {
    private val executor = Executors.newFixedThreadPool(4)

    fun process(update: Update) {
        executor.submit { updateConsumer.accept(update) }
    }
}
